src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.FeatureCollection.cs

```diff
@@ -7,10 +7,19 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.Internal;
 
-internal sealed partial class QuicStreamContext : IPersistentStateFeature, IStreamDirectionFeature, IProtocolErrorCodeFeature, IStreamIdFeature, IStreamAbortFeature
+internal sealed partial class QuicStreamContext :
+    IPersistentStateFeature,
+    IStreamDirectionFeature,
+    IProtocolErrorCodeFeature,
+    IStreamIdFeature,
+    IStreamAbortFeature,
+    IStreamClosedFeature
 {
+    private readonly record struct CloseAction(Action<object?> Callback, object? State);
+
     private IDictionary<object, object?>? _persistentState;
     private long? _error;
+    private List<CloseAction>? _onClosed;
```

Comment:
Maybe consider naming this something like `_onClosedHandlers` or something?

Reply:
Sure. I'll change it in another PR to avoid rebuilding this one in CI.